lldb/source/Target/Target.cpp

```diff
@@ -2747,6 +2747,14 @@ llvm::Optional<SwiftScratchContextReader> Target::GetSwiftScratchContext(
         log->Printf("not allowed to create a new context\n");
       return nullptr;
     }
+
+    auto type_system_or_err =
+        GetScratchTypeSystemForLanguage(eLanguageTypeSwift, false);
+    if (!type_system_or_err) {
+      llvm::consumeError(type_system_or_err.takeError());
+      return nullptr;
+    }
+
```

Comment:
I understand how this avoids the deadlock, it also opens a miniscule window for a race condition where `type_system_or_err` is torn down by another thread between the call to `GetScratchTypeSystemForLanguage()` and `GetSwiftScratchContextLock().try_lock()`. Maybe this can be alleviated by checking
```
              m_scratch_type_system_map.GetTypeSystemForLanguage(
                  language, this, create_on_demand, compiler_options);
```
after acquiring the lock to make sure we got the current one?

Reply:
and take the latest if we don't have the current one?

Reply:
updated